src/ProjectNeon/Assets/Features/Effects/Events/SpellPerformed.cs

```diff
@@ -0,0 +1,11 @@
+﻿
+public sealed class SpellPerformed
+{
+    public SpellFlatDamageEffect SpellFlatDamageEffect { get; }
+
+    public SpellPerformed(SpellFlatDamageEffect spell)
```

Comment:
If we are going to have abstract events for Spells, they can't depend on concrete spell effects.

Reply:
@SilasReinagel Fixed it, made it to depend on generic Effect